server/controllers/documents.js

```diff
@@ -0,0 +1,237 @@
+  req.checkBody('access', 'access cannot be empty').notEmpty();
+  req.checkBody('access', 'public, private and role are the only allowed acces types')
+    .isIn(['public', 'private', 'role']);
+  req.checkBody('access', 'only letters of the alphabets are allowed as access').isAlpha();
+
+  returnValidationErrors(req, res);
+
+  Documents.create({
+    title: req.body.title,
+    content: req.body.content,
+    access: req.body.access,
+    userId: req.decoded.userId,
+  })
+  .then(
+    document => res.status(201).send({
+      status: 'ok',
+      document,
+      message: 'document was successfully created',
+    })
+  )
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description view all documents
+ * @function viewDocument
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const viewDocument = (req, res) => {
+  let offset = 0, limit = 20;
+  if (req.query.limit || req.query.offset) {
+    req.checkQuery('limit', 'Limit must be an integer').isInt();
+    req.checkQuery('offset', 'Offset must be an integer').isInt();
+
+    returnValidationErrors(req, res);
+
+    /** convert limit and offset to number in base 10 */
+    limit = parseInt(req.query.limit, 10);
+    offset = parseInt(req.query.offset, 10);
+  }
+
+  Documents.findAll({
+    include: [
+      {
+        model: Users,
+        required: true,
+        attributes: ['userId', 'fullname'],
+      }
+    ],
+    offset,
+    limit,
+    attributes: { exclude: ['content', 'userId'] },
+  })
+  .then((documents) => {
+    if (documents.length < 1) {
+      return res.status(200).send({
+        status: 'ok',
+        message: 'No document found',
+      });
+    }
+
+    return res.status(200).send({
+      status: 'ok',
+      documents,
+    });
+  })
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description view a document using its id
+ * @function getDocumentById
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const getDocumentById = (req, res) => {
+  req.checkParams('id', 'No document id supplied').notEmpty();
+  req.checkParams('id', 'Only integers are allowed as document id').isInt();
+
+  returnValidationErrors(req, res);
+
+  Documents.findOne({
+    where: { documentId: req.params.id },
+    attributes: { exclude: ['userId'] },
+    include: [
+      {
+        model: Users,
+        required: true,
+        attributes: ['userId', 'fullname', 'roleId'],
+        plain: true,
+      }
+    ],
+  })
+  .then((document) => {
+    if (!document) {
+      return res.status(404).send({
+        status: 'error',
+        message: 'This document does not exist or has been previously deleted',
+      });
+    }
+
+    if (
+      document.access === 'public' ||
+      req.decoded.role === 1 ||
+      document.User.userId === req.decoded.userId ||
+      (document.access === 'role' && document.User.roleId === req.decoded.role)
```

Comment:
Is there a reason why this if statement is so convoluted? Maybe look into abstracting into a function.

Another thing, is this check really necessary? Why not have a simpler condition that checks if a user is authorized to view the document and return a `401` status code. Essentially, wrap this part:
```javascript
return res.status(401).send({
  status: 'error',
  message: 'You are not authorized to view this document',
});
```
in an if condition instead (I assume this will only involve checking the document access level)

and move the current code in the `if` block to the bottom as the last expression.